lib/sycamore/sycamore/transforms/map.py

```diff
@@ -62,20 +63,39 @@ def custom_flat_mapping_function(document: Document) -> list[Document]:
     """
 
     def __init__(self, child: Node, *, f: Callable[[Document], list[Document]], **resource_args):
-        super().__init__(child, **resource_args)
-        self._f = f
-
-    def execute(self) -> "Dataset":
-        input_dataset = self.child().execute()
-        if isinstance(self._f, type):
-            ray_callable = generate_flat_map_class(self._f)
-            return input_dataset.flat_map(ray_callable, compute=ActorPoolStrategy(size=1), **self.resource_args)
+        super().__init__(child, f=FlatMap.wrap(f), name=f.__name__, **resource_args)
+
+    @staticmethod
+    def wrap(f: Callable[[Document], list[Document]]) -> Callable[[list[Document]], list[Document]]:
+        if isinstance(f, type):
+
+            class _Wrap(f):  # type: ignore[valid-type,misc]
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+
+                def __call__(self, docs):
+                    s = super()
+                    ret = []
+                    for d in docs:
+                        ret.extend(s.__call__(d))
+                    return ret
```

Comment:
potential one-liner alternatives:
```python
return sum((s.__call__(d) for d in docs), start=[])

return list(itertools.chain.from_iterable(s.__call__(d) for d in docs)))
```

Reply:
nvm, seems like the manual loop is actually faster

Reply:
Ack.